compiler-core/src/parse.rs

```diff
@@ -2167,14 +2167,27 @@ where
                         self.parse_const_record_finish(start, Some(name), upname, end)
                     }
                     Some((_, Token::Name { name: end_name }, end)) => {
-                        let _ = self.next_tok(); // upname
-                        Ok(Some(Constant::Var {
-                            location: SrcSpan { start, end },
-                            module: Some(name),
-                            name: end_name,
-                            constructor: None,
-                            typ: (),
-                        }))
+                        if self.peek_tok1() == Some(&Token::LeftParen) {
+                            let _ = self.next_tok(); // name
+
+                            parse_error(
+                                ParseErrorType::UnexpectedToken {
```

Comment:
I think a dedicated error would be best here as we want more information, and the `expected` field below isn't quite correct. If we do this we wouldn't need to have the `hint` be part of the error value, it could be added when printing the error.